### PR TITLE
fix: change broken link pointing to semver syntax

### PR DESCRIPTION
**Pull request by lkmandy**
In the generator docs for the configuration generator, the link
pointing to the semver syntax is broken. The correct link
(https://docs.npmjs.com/cli/v6/using-npm/semver) which this broken
link should point to is now version dependent. This means it might
soon become broken with future version updates.

Change the link pointing to the semver syntax, to a beginner-friendly
semver article hosted on the learn section of the nodejs website.

Add a read more link that points to a summarised semver article
hosted on the npm website; which contains more resources and a video
demo.

@jonaslagoni @derberg 
Fixes issue #691

### CODE REVIEW

File: docs/authoring.md [hunk 1]
```diff
@@ -184,7 +184,7 @@ The `generator` property from `package.json` file must contain a JSON object tha
...
 |`nonRenderableFiles`| [String] | A list of file paths or [globs](https://en.wikipedia.org/wiki/Glob_(programming)) that must be copied "as-is" to the target directory, i.e., without performing any rendering process. This is useful when you want to copy binary files.
-|`generator`| [String] | A string representing the Generator version-range the template is compatible with. This value must follow the [semver](https://docs.npmjs.com/misc/semver) syntax. E.g., `>=1.0.0`, `>=1.0.0 <=2.0.0`, `~1.0.0`, `^1.0.0`, `1.0.0`, etc.
+|`generator`| [String] | A string representing the Generator version-range the template is compatible with. This value must follow the [semver](https://nodejs.dev/learn/semantic-versioning-using-npm) syntax. E.g., `>=1.0.0`, `>=1.0.0 <=2.0.0`, `~1.0.0`, `^1.0.0`, `1.0.0`, etc. [read more about semver](https://docs.npmjs.com/about-semantic-versioning)
```
Comment (derberg) [inline docs/authoring.md:187]:
```suggestion
|`generator`| [String] | A string representing the Generator version-range the template is compatible with. This value must follow the [semver](https://nodejs.dev/learn/semantic-versioning-using-npm) syntax. E.g., `>=1.0.0`, `>=1.0.0 <=2.0.0`, `~1.0.0`, `^1.0.0`, `1.0.0`, etc. [Read more about semver](https://docs.npmjs.com/about-semantic-versioning).
```

I think it is better to start it as another sentence, with dot at the end

Comment (lkmandy) [reply docs/authoring.md]: I think it looks better this way too. I just updated the PR.

**Review by derberg — APPROVED** [review]
Thanks so much for well done research and a fix 🙇🏼

**Comment by lkmandy** [comment]
Yay!! It's my first PR at AsyncAPI. Thanks for your guidance all through.

**Comment by derberg** [comment]
@lkmandy thanks for spending your time on the project 🙏🏼

**Comment by derberg** [comment]
@all-contributors please add @lkmandy for doc

**Pull request merged**